perllib/FixMyStreet/App/Controller/Report/New.pm

```diff
@@ -1518,10 +1518,55 @@ sub generate_map : Private {
 sub check_for_category : Private {
     my ( $self, $c ) = @_;
 
-    my $category = $c->get_param('category') || '';
+    my $category = $c->get_param('category') || $c->stash->{report}->category || '';
     $category = '' if $category eq _('Loading...') || $category eq _('-- Pick a category --');
-    $c->stash->{category} = $category || $c->stash->{report}->category || '';
+    $c->stash->{category} = $category;
 
+    if ($c->get_param('submit_category_part_only')) {
+        # If we've clicked the first-part category button (no-JS only probably),
+        # we only want to reshow the form
+        $c->stash->{force_form_not_submitted} = 1;
+    }
```

Comment:
This does mean if they change their mind once the full form is displayed they can still submit in a blocked category. However, that's possibly a future improvement.

Reply:
I've committed 87bba912063d2f2e6b884be58cfc8ffb709a5028 as a fixup which should deal with this - if there's a disable form message to be shown, it knows it shouldn't allow full form submission.